git_theta/updates/base.py

```diff
@@ -0,0 +1,68 @@
+"""Base class for parameter update plugins."""
+
+import os
+import sys
+
+if sys.version_info < (3, 10):
+    from importlib_metadata import entry_points
+else:
+    from importlib.metadata import entry_points
+
+import logging
+from typing import Optional
+
+from git_theta import git_utils, utils, params, metadata
+
+
+class Update:
+    """Base class for parameter update plugins."""
+
+    def read(self, param_metadata):
+        lfs_pointer = param_metadata.lfs_metadata.lfs_pointer
+        serialized_param = git_utils.git_lfs_smudge(lfs_pointer)
+        param = params.get_update_serializer().deserialize(serialized_param)
+        return param
+
+    def get_last_version(self, repo, path, param_keys, param_metadata):
+        last_commit = param_metadata.theta_metadata.last_commit
+        logging.debug(f"Getting data from commit {last_commit}")
+        if last_commit:
+            last_metadata_obj = git_utils.get_file_version(repo, path, last_commit)
+            last_metadata = metadata.Metadata.from_file(last_metadata_obj.data_stream)
+            last_param_metadata = last_metadata.flatten()[param_keys]
+            return last_param_metadata
+        else:
+            raise ValueError("Cannot find previous version for parameters")
+
+    @property
+    def name(self):
+        raise NotImplementedError
+
+    def apply(self, repo, path, param_keys, param_metadata):
+        raise NotImplementedError
+
+    def calculate_update(self, repo, path, param_keys, param_metadata, param):
+        raise NotImplementedError
+
+
+def get_update_name(update_type: Optional[str] = None) -> str:
+    return update_type or os.environ.get(utils.EnvVarConstants.UPDATE_TYPE) or "dense"
+
+
+def get_update(update_type: Optional[str] = None) -> Update:
```

Comment:
You should rebase on main or copy some of the changes over, for example, this has been renamed `get_update_handler` to be more in-line with the checkpoint one.

Reply:
Manually copied/checked-out the stuff that's been incorporated into main since this branch was started.